### PR TITLE
adds: Dynamic Banner Images from Unsplash Collections API

### LINKED ISSUE #802: Use multiple images for our hero banner
**What would you like to be added**:

Let's consider including a bunch of images in our tree that we use (randomly) for the hero banner background image.  We'd still have only 1 image showing (I don't mean rotate through them when you load), but every time you reload the site, you'd get a new one.

Here are some queries with images to consider:

- https://unsplash.com/s/photos/telescope
- https://unsplash.com/s/photos/space
- https://unsplash.com/s/photos/blog
- https://unsplash.com/s/photos/code
- https://unsplash.com/s/photos/developer
- https://unsplash.com/s/photos/technology
- https://unsplash.com/s/photos/learn
- https://unsplash.com/s/photos/write

I'm not sure how many we'd want to have, but "more than one."  We can probably do some processing (manual or via Gatsby plugins) to build all the right sizes https://help.groupthought.com/article/24-images-that-work-well-with-multiple-device-sizes.

This bug is really a 2 (or more) issue:

- decide if we want to do this at all
- pick images we want to use
- integrate them into our Gatsby build so sizing works well for all devices
- add code to the frontend to deal with showing a different each time
- figure out how to do attribution, since we're using CC licensed images

**Why would you like this to be added**:

I really like the image we have, but I also look at it over and over again, and it's losing its interest with repeated expose.  I think if I got a different image every time I went to the site, I might be more engaged with the hero banner, and not want to move past it immediately.

I also think the hero banner image could be another subtle way for us to communicate Telescope's culture and meaning to users.  Over time, and through repeated visits, we could start to seed different ideas about what we're doing here.

I'd love to showcase things like writing, openness, diversity, technology, etc. For example, some random ones to give you a sense (I'm not suggesting we use these, just want to show a bunch of different ideas):

This is somewhat connected to #789

### PR DESCRIPTION
**Pull request by raygervais**
## Issue This PR Addresses
Fixes #802 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [X] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [X] **UI**: Change which improves UI

## Description

This adds background image source generation from Unsplash Collections API. Currently leveraging 
the following collection: https://unsplash.com/collections/894/earth-%26-planets. Pulls images based on viewport width / height, in attempt to ensure image isn't scale oddly between devices. This also generates a unique experience for each device that you access Telescope on.

I plan on adding credit to the collection author before having this PR merged, along with optimization where possible. 

### Examples

image

image

image

image

## Checklist

- [X] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)

- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/frontend/src/components/Banner/Banner.jsx [hunk 1]
```diff
@@ -96,6 +96,58 @@ function ScrollDown(props) {
...
+        // Uses https://unsplash.com/collections/894/earth-%26-planets collection
+        /* Other Options: 
+        - https://unsplash.com/collections/2411320/trend%3A-extreme-neon
+        - https://unsplash.com/collections/1538150/milkyway
+        - https://unsplash.com/collections/291422/night-lights
+        */
+
+        // Ensure we are using an image which fits correctly to user's viewspace
+        const dimensions = `${window.innerWidth}x${window.innerHeight}`;
+        const response = await fetch(`https://source.unsplash.com/collection/894/${dimensions}/`);
```
[the remark this answers is not in the document]
Comment (raygervais) [reply src/frontend/src/components/Banner/Banner.jsx]: Considering the scale of Unsplash, I don't think we are causing any issues by hitting this API per load; it is an open API. That being said, caching would be a better idea if we accepted that there is surprises in the images utilized. Kinda removes the magic in my opinion.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 2]
```diff
@@ -96,6 +96,58 @@ function ScrollDown(props) {
...
+      try {
+        // Uses https://unsplash.com/collections/894/earth-%26-planets collection
+        /* Other Options: 
+        - https://unsplash.com/collections/2411320/trend%3A-extreme-neon
+        - https://unsplash.com/collections/1538150/milkyway
+        - https://unsplash.com/collections/291422/night-lights
+        */
+
+        // Ensure we are using an image which fits correctly to user's viewspace
+        const dimensions = `${window.innerWidth}x${window.innerHeight}`;
```
[the remark this answers is not in the document]
Comment (raygervais) [reply src/frontend/src/components/Banner/Banner.jsx]: Agreed, the idea behind this is actually to tell Unsplash what image dimensions we want to filter on, allowing for someone on a 4K monitor to receive an image closer to 4K res and likewise. In my mind, the idea behind this was to enable sharp images (granted that the original may not be, and that's beyond our control) for `Banner.jsx` based on viewport.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 3]
```diff
@@ -96,6 +96,58 @@ function ScrollDown(props) {
   );
 }
 
+function RetrieveBackgroundImage() {
+  const [backgroundImgSrc, setBackgroundImgSrc] = useState('../../images/hero-banner.png');
```
[the remark this answers is not in the document]
Comment (raygervais) [reply src/frontend/src/components/Banner/Banner.jsx]: Fallback / default image, if you wanted us to utilize the Unsplash image alone (and fall back if needed to this one), then the user would see the dark-gray solid background for a second as the site loads the image.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 4]
```diff
@@ -96,6 +96,58 @@ function ScrollDown(props) {
...
+        }
+
+        const src = await response.url;
+        setTransitionBackground(true);
+
+        // Ease in Background
+        setTimeout(() => {
+          setBackgroundImgSrc(src);
+          setTransitionBackground(false);
+        }, 1000);
```
[the remark this answers is not in the document]
Comment (raygervais) [reply src/frontend/src/components/Banner/Banner.jsx]: This was simply UX magic, to have a smooth transition between images. If we are not using the old one unless error-ed out, then I can remove this entirely.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 5]
```diff
@@ -96,6 +96,58 @@ function ScrollDown(props) {
...
+      } catch (error) {
+        console.error('Error getting user info', error);
```
[the remark this answers is not in the document]
Comment (raygervais) [reply src/frontend/src/components/Banner/Banner.jsx]: Agreed, updated :computer:

**Comment by raygervais** [comment]
Can I get some eyes on this @humphd @agarcia-caicedo when you have a moment

**Comment by humphd** [comment]
I'm tempted to do any major changes in follow-ups vs. blocking this, but I'd like to hear from Ana first. @agarcia-caicedo can you give this a final once-over?

**Comment by humphd** [comment]
@agarcia-caicedo great, thanks.  @raygervais when you're ready with this, go ahead and merge.  We'll do follow-up work in new PRs.

**Pull request merged**